src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -705,21 +705,23 @@ def inner(*args, **kwargs):
             kwargs.update(params)
 
             if version is not None:
+                exp_version = APIVersion.to_string(version)
                 accept_header = cherrypy.request.headers.get('Accept')
-                if accept_header and accept_header.startswith('application/vnd.ceph.api.v'):
-                    req_match = re.search(r"\d\.\d", accept_header)
-                    if req_match:
-                        req_version = req_match[0]
+                req_match = APIVersion.from_string(accept_header)
+                if accept_header and req_match:
+                    req_version = req_match
                 else:
                     raise cherrypy.HTTPError(415, "Unable to find version in request header")
 
-                if req_version and req_version == version:
+                if APIVersion(exp_version.major, exp_version.minor).handled_by(req_version):
```

Comment:
exp_version is a APIVersion object (despite the funny "to_string" factory method):
```suggestion
                if exp_version.handled_by(req_version):
```